qqq-backend-module-rdbms/src/main/java/com/kingsrook/qqq/backend/module/rdbms/jdbc/QueryManager.java

```diff
@@ -526,18 +529,57 @@ public static Integer executeInsertForGeneratedId(Connection connection, String
    /*******************************************************************************
     ** todo - needs (specific) unit test
     *******************************************************************************/
-   public static List<Integer> executeInsertForGeneratedIds(Connection connection, String sql, List<Object> params) throws SQLException
+   public static List<Serializable> executeInsertForGeneratedIds(Connection connection, String sql, List<Object> params, Map<String, QFieldMetaData> fields) throws SQLException
    {
       try(PreparedStatement statement = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS))
       {
          bindParams(params.toArray(), statement);
          incrementStatistic(STAT_QUERIES_RAN);
          statement.executeUpdate();
-         ResultSet     generatedKeys = statement.getGeneratedKeys();
-         List<Integer> rs            = new ArrayList<>();
+
+         /////////////////////////////////////////////////////////////
+         // We default to idType of INTEGER but if we are passed in //
+         // fields attempt to find the dataType of the id field     //
+         /////////////////////////////////////////////////////////////
+         QFieldType idType = QFieldType.INTEGER;
+         if(fields != null && !fields.isEmpty())
+         {
+            Optional<QFieldMetaData> field = fields.values().stream()
+               .filter(f -> f.getName().equals("id"))
+               .findFirst();
+
+            if(field.isPresent())
+            {
+               /////////////////////////////////////////////////////////
+               // if we find "id" field get the type and use it below //
+               /////////////////////////////////////////////////////////
+               idType = field.get().getType();
+            }
+         }
+
+         ResultSet          generatedKeys = statement.getGeneratedKeys();
+         List<Serializable> rs            = new ArrayList<>();
          while(generatedKeys.next())
          {
-            rs.add(getInteger(generatedKeys, 1));
+            switch(idType)
+            {
+               case INTEGER:
+               {
+                  rs.add(getInteger(generatedKeys, 1));
+                  break;
+               }
+               case LONG:
+               {
+                  rs.add(getLong(generatedKeys, 1));
+                  break;
+               }
+               default:
+               {
+                  LOG.warn("Unknown id data type, attempting to getInteger.", logPair("sql", sql));
```

Comment:
/if/ we had a table with a non-integer or long id, this would log A TON (e.g., for every insert)...  

one *might* keep a Set of the SQL strings that we have logged about, and only warn once per sql string?  idk, probably will never come up so not worth changing until it does come up.